atc/atccmd/command.go

```diff
@@ -96,387 +99,252 @@ const algorithmLimitRows = 100
-		EnableContainerAuditLog bool `long:"enable-container-auditing" description:"Enable auditing for all api requests connected to containers."`
-		EnableJobAuditLog       bool `long:"enable-job-auditing" description:"Enable auditing for all api requests connected to jobs."`
-		EnablePipelineAuditLog  bool `long:"enable-pipeline-auditing" description:"Enable auditing for all api requests connected to pipelines."`
-		EnableResourceAuditLog  bool `long:"enable-resource-auditing" description:"Enable auditing for all api requests connected to resources."`
-		EnableSystemAuditLog    bool `long:"enable-system-auditing" description:"Enable auditing for all api requests connected to system transactions."`
-		EnableTeamAuditLog      bool `long:"enable-team-auditing" description:"Enable auditing for all api requests connected to teams."`
-		EnableWorkerAuditLog    bool `long:"enable-worker-auditing" description:"Enable auditing for all api requests connected to workers."`
-		EnableVolumeAuditLog    bool `long:"enable-volume-auditing" description:"Enable auditing for all api requests connected to volumes."`
-	}
-
-	Syslog struct {
-		Hostname      string        `long:"syslog-hostname" description:"Client hostname with which the build logs will be sent to the syslog server." default:"atc-syslog-drainer"`
-		Address       string        `long:"syslog-address" description:"Remote syslog server address with port (Example: 0.0.0.0:514)."`
-		Transport     string        `long:"syslog-transport" description:"Transport protocol for syslog messages (Currently supporting tcp, udp & tls)."`
-		DrainInterval time.Duration `long:"syslog-drain-interval" description:"Interval over which checking is done for new build logs to send to syslog server (duration measurement units are s/m/h; eg. 30s/30m/1h)" default:"30s"`
-		CACerts       []string      `long:"syslog-ca-cert"              description:"Paths to PEM-encoded CA cert files to use to verify the Syslog server SSL cert."`
-	} ` group:"Syslog Drainer Configuration"`
-
-	Auth struct {
-		AuthFlags     skycmd.AuthFlags
-		MainTeamFlags skycmd.AuthTeamFlags `group:"Authentication (Main Team)" namespace:"main-team"`
-	} `group:"Authentication"`
-
-	ConfigRBAC flag.File `long:"config-rbac" description:"Customize RBAC role-action mapping."`
+	CredentialManagement creds.CredentialManagementConfig `yaml:"credential_management,omitempty"`
+	CredentialManager    string                           `yaml:"credential_manager,omitempty" validate:"creds_manager"`
+	CredentialManagers   CredentialManagersConfig         `yaml:",inline"`
 
-	SystemClaimKey    string   `long:"system-claim-key" default:"aud" description:"The token claim key to use when matching system-claim-values"`
-	SystemClaimValues []string `long:"system-claim-value" default:"concourse-worker" description:"Configure which token requests should be considered 'system' requests."`
+	ComponentRunnerInterval time.Duration `yaml:"component_runner_interval,omitempty"`
```

Comment:
Will this successfully parse a string duration value like `10s`?

Reply:
I haven't tested it yet but it is just using `yaml.Unmarshal` or `yaml.Decode` (or pflags) to populate that so it should work? I'll confirm once I have tested it though.